server/plugin.go

```diff
@@ -82,10 +82,17 @@ func (p *Plugin) MessageWillBePosted(c *plugin.Context, post *model.Post) (*mode
 
 		if origText != "" {
 			newText := origText
+
+			channel, _ := p.API.GetChannel(post.ChannelId)
+			team, _ := p.API.GetTeam(channel.TeamId)
+
 			for _, l := range links {
-				newText = l.Replace(newText)
+				if len(l.link.Scope) == 0 {
+					newText = l.Replace(newText)
+				} else if contains(channel.Name, team.Name, l.link.Scope) {
```

Comment:
It's kind of hard to read. That do you think about using 
```suggestion
				if len(l.link.Scope) != 0 && !contains(channel.Name, team.Name, l.link.Scope) {
					continue
				}
				newText = l.Replace(newText)
```

instead?

Reply:
If there is no Scope provided in configuration, == 0 is a valid check.

Let me know if I missed something.

Reply:
Maybe we misunderstand each other.

I'm proposing to check first, if a scope is set and then if the scope is meet. The default case should be to apply the replacer.

Reply:
Will you block on this?

Reply:
If you are at least 4/5, that you code is well the current way, no. I'm just 2/5.

Reply:
I feel that handling 

No Scope in configuration
or
Scope contains team & channel
or 
Scope contains team

I am 4/5 I would change it if you found a more elegant solution to the above logic.

Reply:
Could you please point out, what is wrong on the solution pointed out above?

Reply:
Sorry, but now I want ambitious to figure out "the right solution".